Review request for RecoveryServicesSiteRecovery to add version stable/2024-10-01

singhabh27 opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

singhabh27 committed:
﻿Copy files from stable/2024-04-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

singhabh27 committed:
﻿Update version to stable/2024-10-01

Updated the API version from stable/2024-04-01 to stable/2024-10-01.

singhabh27 committed:
﻿Added tag for 2024-10-01 in readme file

singhabh27 committed:
October API Version changes

singhabh27 committed:
Swagger PrettierCheck fix

prsadhu-ms-idc commented:
/pr DoNotMerge

mentat9 commented:
>         "linuxLicenseType": {

It looks like 6 copies of the same definition. Not a requirement, but it might be easier to maintain if you shared one definition in 6 places.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-10-01/service.json:19714 in dd768cf. [](commit_id = dd768cf1eee6f6096507a8f834c604dcb556a46f, deletion_comment = False)

msyyc commented on specification/recoveryservicessiterecovery/resource-manager/readme.md:
````diff
@@ -72,6 +72,15 @@ directive:
+
+```yaml $(tag) == 'package-2024-10'
+input-file:
+  - Microsoft.RecoveryServices/stable/2024-10-01/service.json
````
All the `"x-ms-parameter-location": "client"` in service.json shall be removed otherwise it will cause unexpected breakings for SDK.

singhabh27 replied:
Removed "x-ms-parameter-location": "client"

msyyc replied:
I already added it in readme.python.md so you don't need remove it for python now.

singhabh27 committed:
Taking up comments

singhabh27 commented:
/pr DoNotMerge

tadelesh commented:
/azp run

kazrael2119 commented:
please add "x-ms-parameter-location": "method" for these global parameters to avoid breaking：
https://github.com/Azure/azure-rest-api-specs/blob/fd9c9ba499462f099ef0fd4a316c668d0bb27bc9/specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-10-01/service.json#L28091-L28104
and
https://github.com/Azure/azure-rest-api-specs/blob/fd9c9ba499462f099ef0fd4a316c668d0bb27bc9/specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-10-01/service.json#L28112-L28135

msyyc commented:
> please add "x-ms-parameter-location": "method" for these global parameters to avoid breaking：
> 
> https://github.com/Azure/azure-rest-api-specs/blob/fd9c9ba499462f099ef0fd4a316c668d0bb27bc9/specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-10-01/service.json#L28091-L28104
> 
> 
> and
> https://github.com/Azure/azure-rest-api-specs/blob/fd9c9ba499462f099ef0fd4a316c668d0bb27bc9/specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-10-01/service.json#L28112-L28135

default location is "method" so there is no need to add it.

Alancere commented:
/azp run

msyyc commented:
/azp run

msyyc committed:
Merge branch 'main' into users/singhabh27/Oct24ApiVersion

singhabh27 commented:
@msyyc, please add "Do not merge" label, somehow unable to add labels in the PR.

msyyc commented:
> @msyyc, please add "Do not merge" label, somehow unable to add labels in the PR.

I am python SDK owner and there is no breaking change for Python SDK of this PR.

msyyc commented:
> @msyyc, just as a confirmation, do I need to add "x-ms-parameter-location": "client" back in service.json?

Python already add directive in `readme.python.md` of https://github.com/Azure/azure-rest-api-specs/commit/d460585857947d5f15fe022ac5a8b2b49f6e0761 so Python doesn't need it now. @kazrael2119/ @qiaozha  is JS SDK owner and you can contact them whether they need you to add in swagger.

kazrael2119 commented:
@singhabh27 , the default value of x-ms-parameter-location is client so you don't need to add client back, just need do this  is all right
> please add "x-ms-parameter-location": "method" for these global parameters to avoid breaking：
> 
> https://github.com/Azure/azure-rest-api-specs/blob/fd9c9ba499462f099ef0fd4a316c668d0bb27bc9/specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-10-01/service.json#L28091-L28104
> 
> 
> and
> https://github.com/Azure/azure-rest-api-specs/blob/fd9c9ba499462f099ef0fd4a316c668d0bb27bc9/specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-10-01/service.json#L28112-L28135

singhabh27 committed:
Adding UserSelectedOSName and taking up comments.

singhabh27 committed:
Merge branch 'users/singhabh27/Oct24ApiVersion' of https://github.com/singhabh27/azure-rest-api-specs into users/singhabh27/Oct24ApiVersion

singhabh27 commented:
@kazrael2119, please add WaitForARMFeedback label.

singhabh27 commented:
/azp run

singhabh27 commented:
/azp run

prsadhu-ms-idc commented:
/azp run

singhabh27 committed:
Reverting x-ms-parameter-location changes and add it only for global parameters

singhabh27 committed:
Fixing Lintdiff

singhabh27 committed:
Reverting last 4 commits

singhabh27 committed:
Adding userSelectedOSName property in HyperVReplicaAzureUpdateReplicationProtectedItemInput

singhabh27 committed:
Added "x-ms-parameter-location": "method" for global parameters

singhabh27 commented:
@kazrael2119 adding "x-ms-parameter-location": "method" is causing breaking change in the PR. How to resolve the error?

singhabh27 committed:
Changing to "x-ms-parameter-location": "path"

singhabh27 committed:
Merge branch 'main' into users/singhabh27/Oct24ApiVersion

singhabh27 committed:
Revert "Changing to "x-ms-parameter-location": "path""

This reverts commit 7ef097272c6aa3ba58318bd7dd1eb73a06123b21.

singhabh27 committed:
Merge branch 'users/singhabh27/Oct24ApiVersion' of https://github.com/singhabh27/azure-rest-api-specs into users/singhabh27/Oct24ApiVersion

singhabh27 committed:
Fixing Lint diff

prsadhu-ms-idc committed:
Added nic name in update and get call objects

weidongxu-microsoft commented:
Appears no breaking change detected in Java.

qiaozha commented:
I remember we don't require breaking change board review before for cross version breaking change? @mikekistler 
If it is, I think this x-ms-parameter-location change doesn't impact the rest api layer ? and it should be the right thing for SDK client side.

singhabh27 commented:
@qiaozha  / @mikekistler, pl. remove BreakingChangeReviewRequired tag and NotReadyForArmReview tag, so that ARM reviewers to review the PR.

singhabh27 committed:
Adding userSelectedOSName for CBT and V2A-RCM

rkmanda reviewed (approved):
:shipit:

anmolbhatia289 committed:
Added addIdentity API

Added add identity API for tenant migration

anmolbhatia289 committed:
running prettier commands

anmolbhatia289 committed:
Removed ServiceDefaultError

anmolbhatia289 committed:
Removing required from api-version, resource name, RG name and subscription Id and adding fixing example version

Removing required from api-version, resource name, RG name and subscription Id and adding fixing example version

anmolbhatia289 committed:
Adding required instance type + re-oredering

anmolbhatia289 committed:
adding location, defaults and fix path camel case

anmolbhatia289 committed:
adding pattern to resourceName

anmolbhatia289 committed:
adding other parameters for pattern

anmolbhatia289 committed:
Adding directly

anmolbhatia289 committed:
fixing example.json

anmolbhatia289 committed:
Removing add Identity API

prsadhu-ms-idc committed:
Added OS upgrade inputs"

prsadhu-ms-idc committed:
Merge branch 'users/singhabh27/Oct24ApiVersion' of https://github.com/singhabh27/azure-rest-api-specs into users/singhabh27/Oct24ApiVersion

ramoka178 commented:
>     "InMageRcmUnplannedFailoverInput": {

InMageRcmUnplannedFailoverInput and InMageRcmTestFailoverInput where the recent commit had 'osUpgradeVersion' added is not being used as a definition. 

Could you check if its needed? and also update examples reflecting this field 'osUpgradeVersion' ?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-10-01/service.json:19579 in acad3b7. [](commit_id = acad3b705f09d2790747b10bee2f2f014e498a9c, deletion_comment = False)

singhabh27 committed:
Updated description of Recoveryplan reprotect.

prsadhu-ms-idc committed:
Added examples for OS upgrade in VMware CBT test migration

prsadhu-ms-idc committed:
Merge branch 'users/singhabh27/Oct24ApiVersion' of https://github.com/singhabh27/azure-rest-api-specs into users/singhabh27/Oct24ApiVersion

prsadhu-ms-idc commented:
> > ```
> > "InMageRcmUnplannedFailoverInput": {
> > ```
> 
> InMageRcmUnplannedFailoverInput and InMageRcmTestFailoverInput where the recent commit had 'osUpgradeVersion' added is not being used as a definition.
> 
> Could you check if its needed? and also update examples reflecting this field 'osUpgradeVersion' ?
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2024-10-01/service.json:19579 in [acad3b7](https://github.com/Azure/azure-rest-api-specs/commit/acad3b705f09d2790747b10bee2f2f014e498a9c). [](commit_id = [acad3b7](https://github.com/Azure/azure-rest-api-specs/commit/acad3b705f09d2790747b10bee2f2f014e498a9c), deletion_comment = False)

InMageRcmUnplannedFailoverInput and InMageRcmTestFailoverInput are referred to by their base classes in other API inputs. As for example, I have created a new file for same.

singhabh27 committed:
Merge branch 'main' into users/singhabh27/Oct24ApiVersion

prsadhu-ms-idc committed:
Removed new example file and added OS upgrade example in existing file

prsadhu-ms-idc committed:
Merge branch 'users/singhabh27/Oct24ApiVersion' of https://github.com/singhabh27/azure-rest-api-specs into users/singhabh27/Oct24ApiVersion

prsadhu-ms-idc committed:
Ran prettier check on updated file
